src/statisticBlock.js

```diff
@@ -111,13 +109,35 @@ class StatisticBlock extends Component {
       let clientWidth = this.state.clientWidth;
       let clientHeight = this.state.clientHeight;
       let childHeight = 0;
+      let childWidth = 0;
+      let containerElm = this.refs.parent;
+      let childrenElm =containerElm.children;
+      let childrenCombinedWidth = 0;
+      let childrenHeight = childrenElm[0].clientHeight;
+      for (let e of childrenElm) {
+        childrenCombinedWidth = childrenCombinedWidth + e.clientWidth;
+      }
 
-      if(element.clientHeight == element.scrollHeight
-      && this.state.size == this.props.options.size
-      && !this.state.overflow) return;
-
+      // if(element.clientHeight == element.scrollHeight
+      // && this.state.size == this.props.options.size
+      // && !this.state.overflow) return;
+
+      if (elementClientWidth < childrenCombinedWidth || elementClientHeight < childrenHeight){
+        let index = getSizeIndex(size);
+        if(index > 0) {
+          // trying to reduce size ...
+          this.setState({
+            size: SIZE_OPTIONS[index - 1].value,
+            clientWidth: elementClientWidth,
+            clientHeight: elementClientHeight,
+            prevClientWidth: this.state.clientWidth,
+            prevClientHeight: this.state.clientHeight
+          });
+        }
+      }
       if(this.refs['child-0']) {
         childHeight = ReactDOM.findDOMNode(this.refs['child-0']).clientHeight;
+        childWidth = ReactDOM.findDOMNode(this.refs['child-0']).clientWidth;
```

Comment:
Traversing a DOM tree is expensive., this could be optimized to only doing a single lookup.
```js
const element = ReactDOM.findDOMNode(this.refs['child-0']);
const { childHeight, clientWidth } = element;
```